### ISSUE TITLE
[Protocol] MVP implementation off the on-chain `CreateClaim` message handling

### ISSUE THREAD

**Issue by Olshansk** [pokt-network/poktroll#140]
## Objective

MVP Implementation of `CreateClaim` to enable Shannon TestNet.

## Origin Document

- #43

## Goals

- Enable E2E validation of the entire protocol
- Unblock iterative development of the claim & proof lifecycle
- Unblock iterative development of on-chain tokenomics

## Deliverables

- [x] A single PR that introduces the business logic for the on-chain `CreateClaim` message (verification, etc...)
- [x] A single PR that updates the CLI to submit CreateClaim messages
- [x] Introduce make targets to enable development (visibility, triggering, debugging , etc...) of on-chain claims
- [ ] Add extensive unit tests 

## Non-goals / Non-deliverables

- Account Management (i.e. Burn & Mint) business logic related to the Claim & Proof lifecycle (leave TODOs)
- Governance Params necessary for the Claim & Proof lifecycle (leave TODOs)

## General deliverables

- [x] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [x] **Testing**: Add new tests (unit and/or E2E) to the test suite.
- [ ] **Makefile**: Add new targets to the Makefile to make the new functionality easier to use.
- [ ] **Documentation**: Update architectural or development READMEs; use [mermaid](https://mermaid-js.github.io/mermaid/) diagrams where appropriate.

---

**Creator**: @Olshansk 
**Co-Owners**: @bryanchriswhite

**Comment by Olshansk** [comment]
@bryanchriswhite Picking this up

**Comment by Olshansk** [comment]
@bryanchriswhite I'm unassigning myself from this in the meantime and will let you pick it up.

I think the comments you left (copy-pastedbelow) in `func (k msgServer) CreateClaim(goCtx context.Context, msg *types.MsgCreateClaim) ` is a great starting point!

```go
	/*
		TODO_INCOMPLETE: Handling the message

		## Validation

		### Session validation
		1. [ ] claimed session ID matches on-chain session ID
		2. [ ] this supplier is in the session's suppliers list

		### Msg distribution validation (depends on session validation)
		1. [ ] governance-based earliest block offset
		2. [ ] pseudo-randomize earliest block offset

		### Claim validation
		1. [ ] session validation
		2. [ ] msg distribution validation

		## Persistence
		1. [ ] create claim message
			- supplier address
			- session header
			- claim
		2. [ ] last block height commitment; derives:
			- last block committed hash, must match proof path
			- session ID (?)

**Comment by Olshansk** [comment]
@bryanchriswhite I'm closing this out and we'll create new tickets to capture any missing work.

**Comment by bryanchriswhite** [comment]
@Olshansk I had the same thought and opened #362. :+1:
